lib/cypress/expected_results_calculator.rb

```diff
@@ -73,21 +76,69 @@ def aggregate_results_for_measure(measure, individual_results = nil)
       end
     end
+
+    # adds the observation values found in an individual_result to the observation_hash
+    def collect_observations(observation_hash, individual_result, measure)
+      key = individual_result['population_set_key']
+      population_set = measure.population_set_for_key(key).first
+      # collect the observation_statments for the population_set. There may be more than one. episode_results are recorded in the same order
+      observation_statments = population_set.observations.map { |obs| obs.observation_parameter.statement_name }
+      # collect the observation_values from and individual_result
+      # a scenario with multiple episodes and multiple observations would look like this [[2], [9, 1]]
+      observation_values = get_observ_values(individual_result['episode_results']).compact
+      observation_values.each do |observation_value|
+        observation_value.each_with_index do |observation, index|
+          # lookup the population code (e.g., DENOM is the population code for the statement named 'Denominator')
+          obs_pop = measure.population_keys.find { |pop| population_set.populations[pop]['statement_name'] == observation_statments[index] }
+          # create an empty hash value if one doesn't already exist
+          observation_hash[key][obs_pop] = { values: [], statement_name: observation_statments[index] } unless observation_hash[key][obs_pop]
+          # add the observation to the hash
+          observation_hash[key][obs_pop][:values] << observation
+        end
+      end
+    end
+
+    def calculate_observation(observation_hash, measure, population_set_key)
+      key = population_set_key
+      return unless observation_hash[key]
+
+      # calculate the aggregate observation based on the aggregation type
+      # aggregation type is looked up using the statement_name
+      observation_hash[key].each do |population, observation_map|
+        pop_set = measure.population_set_for_key(key).first
+        # find observation that matches the statement_name
+        observation = pop_set.observations.select { |obs| obs.observation_parameter.statement_name == observation_map[:statement_name] }.first
+        # Guidance for calculations can be found here
+        # https://www.hl7.org/documentcenter/public/standards/vocabulary/vocabulary_tables/infrastructure/vocabulary/ObservationMethod.html#_ObservationMethodAggregate
+        case observation.aggregation_type
+        when 'COUNT'
+          @measure_result_hash[measure.hqmf_id][key]['observations'][population] = { method: 'COUNT', value: count(observation_map[:values]),
+                                                                                     hqmf_id: observation.hqmf_id }
+        when 'MEDIAN'
+          @measure_result_hash[measure.hqmf_id][key]['observations'][population] = { method: 'MEDIAN', hqmf_id: observation.hqmf_id,
```

Comment:
Can we just put this in the same order as the others for code clarity (hqmf_id last)?